### PR TITLE
ADF-3 Add a hidden field to check if the form has a file upload

### PR DESCRIPTION
**Pull request by chinnu-focaloid**
Relates to: https://oat-sa.atlassian.net/browse/ADF-3

Add a hidden field to check if the form has a file upload

### CODE REVIEW

File: actions/class.Items.php [hunk 1]
```diff
@@ -219,6 +219,15 @@ public function editItem()
             $this->setData('isAuthoringEnabled', $hasModel);
 
             $this->setData('formTitle', __('Edit Item'));
+            
+            if($myForm->hasAsyncFileUpload()){
```
Comment (shpran) [inline actions/class.Items.php:223]:
```suggestion
            if ($myForm->hasAsyncFileUpload()) {
```

File: actions/class.Items.php [hunk 2]
```diff
@@ -219,6 +219,15 @@ public function editItem()
...
+                $hiddenElement = \tao_helpers_form_FormFactory::getElement('', 'hidden');
```
Comment (shpran) [inline actions/class.Items.php:224]: Is it ok that we provide an empty string as the first parameter?

Comment (chinnu-focaloid) [reply actions/class.Items.php]: Yes. Either we can pass as empty string (Which is used to set the name of the element) or pass an argument to set the name
Here I am going to refactor by passing name as the first param and removing the line 
`$hiddenElement->setName('check-file');`

File: actions/class.Items.php [hunk 3]
```diff
@@ -219,6 +219,14 @@ public function editItem()
             $this->setData('isAuthoringEnabled', $hasModel);
 
             $this->setData('formTitle', __('Edit Item'));
+            
+            if ($myForm->hasAsyncFileUpload()) {
```
Comment (gabrielfs7) [inline actions/class.Items.php:223]: We will need this property to work on all sort of extensions that work with metadata. Example. Tests. This fix should be applied on tao-core most probably.

Wdyt @chinnu-focaloid ?

**Review by gabrielfs7 — CHANGES_REQUESTED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: actions/class.Items.php [hunk 4]
```diff
@@ -219,6 +219,7 @@ public function editItem()
             $this->setData('isAuthoringEnabled', $hasModel);
 
             $this->setData('formTitle', __('Edit Item'));
+
```
Comment (boajer) [inline actions/class.Items.php:222]: If that is the only change in whole PR I would recommend just close it ( unless I'm experiencing some strong issue with github cache )

**Review by habemuscode — CHANGES_REQUESTED** [review]
As long as this PR doesn't have any change, just close it and remove the branch. Thanks

**Comment by chinnu-focaloid** [comment]
No chnages in this PR request.

**Pull request closed without merging**